### ISSUE TITLE
Fast refresh si loaded twice

### ISSUE THREAD

**Issue by Relaxe111** [vitejs/vite-plugin-react#193]
### Description

As a developer using vite in an micro front end environment we do need to have multiple loads of the `@react-refresh` when we are in dev mode with multiple micro apps. 
In version 4 was added a `throw error` when one has multiple loads of the react-refresh which we do, because they are located on different urls, while window object is shared with all micro-apps.
 And **window.__vite_plugin_react_runtime_loaded__** is shared as well which will cause a throw, and any micro app after first loaded micro app will die.

<img width="984" alt="CleanShot 2023-07-06 at 11 26 10@2x" src="https://github.com/vitejs/vite-plugin-react/assets/40892030/2dfc32d4-9e35-43c9-80b0-d3baca2b4cc6">

### Suggested solution

Could we add an option  which can disable react-refresh throw `React refresh runtime was loaded twice...`?

```
/* eslint-disable no-undef */
if (typeof window !== 'undefined') {
    **if(!disableThrowMultiLoading){**
      if (window.__vite_plugin_react_runtime_loaded__) {
        throw new Error(
          'React refresh runtime was loaded twice. Maybe you forgot the base path?',
        )
      }
    **}**
  window.__vite_plugin_react_runtime_loaded__ = true
}

```

### Alternative

Alternative can be not to throw but to console.warn  this.

```
/* eslint-disable no-undef */
if (typeof window !== 'undefined') {
 
      if (window.__vite_plugin_react_runtime_loaded__) {
       --// throw new Error(
       ++ console.warn(
          'React refresh runtime was loaded twice. Maybe you forgot the base path?',
        )
      }

  window.__vite_plugin_react_runtime_loaded__ = true
}

```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite-plugin-react/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite-plugin-react/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by ArnaudBarre** [comment]
See discussion here https://github.com/vitejs/vite-plugin-react/discussions/185

How do you manage to load React only one time currently?

**Comment by Relaxe111** [comment]
If I understand correctly you're question, I do load for each project individual React version, with their own /react-refresh. I am using qiankun(single-spa) setup for that, which does runtime assembly of the apps. All micro apps having its owns dev servers, setups, and have its own repos, builds etc. 
Our current workflow is: I am going to a dev url ex: https://dev.advoca.no. and on right down side i have a button, where I can override micro-apps urls with local instances of micro apps running locally, each local micro app in dev mode loads its own React version and /react-refresh, they run independently and can have different versions.

Prior v 4 I can override as much micro-apps as i need, with no problem.

**Comment by ArnaudBarre** [comment]
I released 4.0.3 without this check!

**Comment by Relaxe111** [comment]
thanks a lot!!!
